python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -206,85 +305,116 @@ def _sync(
-                csv_path = "/path/to/file.csv"
+        sling_source = self.source_connection.dict()
+        sling_target = self.target_connection.dict()
 
-                @asset
-                def run_sync(context, sling: SlingResource):
-                    res = sling.sync(
-                        source_stream=csv_path,
-                        target_object="events",
-                        mode=SlingMode.FULL_REFRESH,
-                    )
-                    for stdout in res:
-                        context.log.debug(stdout)
-                    counts = sqlite3.connect(sqllitepath).execute("SELECT count(1) FROM events").fetchone()
-                    assert counts[0] == 3
+        if self.source_connection.connection_string:
+            sling_source["url"] = self.source_connection.connection_string
+        if self.target_connection.connection_string:
+            sling_target["url"] = self.target_connection.connection_string
+        with environ(
+            {
+                "SLING_SOURCE": json.dumps(sling_source),
+                "SLING_TARGET": json.dumps(sling_target),
+            }
+        ):
+            config = {
+                "mode": mode,
+                "source": {
+                    "conn": "SLING_SOURCE",
+                    "stream": source_stream,
+                    "primary_key": primary_key,
+                    "update_key": update_key,
+                    "options": source_options,
+                },
+                "target": {
+                    "conn": "SLING_TARGET",
+                    "object": target_object,
+                    "options": target_options,
+                },
+            }
+            config["source"] = {k: v for k, v in config["source"].items() if v is not None}
+            config["target"] = {k: v for k, v in config["target"].items() if v is not None}
 
-                source = SlingSourceConnection(
-                    type="file",
-                )
-                target = SlingTargetConnection(type="sqlite", instance=sqllitepath)
+            sling_cli = Sling(**config)
+            logger.info("Starting Sling sync with mode: %s", mode)
+            cmd = sling_cli._prep_cmd()  # noqa: SLF001
 
-                materialize(
-                    [run_sync],
-                    resources={
-                        "sling": SlingResource(
-                            source_connection=source,
-                            target_connection=target,
-                            mode=SlingMode.TRUNCATE,
-                        )
-                    },
-                )
+            # `_prep_cmd` only works with the Single Task command, so we need to replace it with the Replication command
+            # TODO: Unfortunately, _prep_cmd doesn't expose a way to add streams, so we'll need to override it ourselves
+            # We'll also need to parse the logs differently, since the output is different
+            # cmd = cmd.replace(" -c ", " -r ")
```

Comment:
Will a single task will still work with the new asset builder?

Reply:
Hmm...
In this PR, single task is what's used.
In the PR up the stack, we move the builder to only use replication.
We _could_ add a flag in the future so users can swap between them, with ack that the logic would be significantly different between the two.

Reply:
The comment was a placeholder (tbh, artifact) for myself for the PR that's upstack. Was trying to keep track of where the right place to swap out the single task for the replication.

'pologies for the confusion with the comment! For all intents and purposes, this PR uses single tasks.